### PR TITLE
Remove legacy state translations

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

These legacy translations were removed in https://github.com/home-assistant/core/pull/112023

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/state/translations-mixin.ts [hunk 1]
```diff
@@ -77,7 +75,15 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
...
       [category: string]: LoadedTranslationCategory;
-    } = {};
+    } = {
+      // state is no longer fetched from the backend but it is
+      // still used in the frontend so we need to initialize it
+      state: {
+        integrations: [],
+        setup: false,
+        configFlow: false,
+      },
```
Comment (silamon) [inline src/state/translations-mixin.ts:85]: Is this added as a precaution or are there specific cases where frontend still uses this?

Comment (bdraco) [reply src/state/translations-mixin.ts:85]: The UI is pretty broken without this.

Comment (silamon) [reply src/state/translations-mixin.ts:85]: Would you mind giving an example? I've applied your code changes with and without this part and UI remains to load in both cases. It sounds like there still somewhere a fallback to state translations that may be applied.

Comment (bdraco) [reply src/state/translations-mixin.ts]: I removed it, and it seems fine today.  Yesterday all the translations wouldn't load without it. Perhaps it was a cache problem or something dirty on my local branch.

I've cleared the cache and tried 3 different browsers and it still works so I'm not sure what has changed between yesterday and today, but happy to not have to have the code if its not needed

**Review by silamon — APPROVED** [review]

File: src/state/translations-mixin.ts [hunk 2]
```diff
@@ -286,23 +284,6 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
...
       force = false
     ): Promise<LocalizeFunc> {
-      if (
-        __BACKWARDS_COMPAT__ &&
-        !atLeastVersion(this.hass!.connection.haVersion, 0, 109)
-      ) {
-        if (category !== "state") {
-          return this.hass!.localize;
-        }
-        const resources = await getHassTranslationsPre109(this.hass!, language);
```
Comment (bramkragten) [inline src/state/translations-mixin.ts:296]: I don't think this should have been removed, it is for when the frontend is not bundled with the backend (cast, supervisor) and is only called when the backed version is old.

Comment (bdraco) [reply src/state/translations-mixin.ts:296]: https://github.com/home-assistant/frontend/pull/20597

**Pull request merged**